src/screens/Habitoptions.js

```diff
@@ -1,132 +1,145 @@
 import React, { useState } from 'react';
-import { View, Text, StyleSheet, TouchableOpacity, FlatList, TextInput, Button } from 'react-native';
+import { View, Text, StyleSheet, TouchableOpacity, ScrollView, TextInput, Button } from 'react-native';
+
+const habitCategories = {
+  'Health and Fitness': ['Exercise', 'Morning Stretching', 'Walk', 'Eat Healthy', 'Cycle', 'Swim', 'Work Out', 'Brush Your Teeth', ],
+  'Personal Growth': ['Read', 'Write in Journal', 'Meditate', 'Limited Screen Time'],
+  'Productivity': ['Complete Daily To-Do List', 'Study'],
+  'Social and Family': ['Spend Time with Family', 'Connect with Friends', 'Express Gratitude'],
+  'Leisure': ['Cooking', 'Outdoor Time', 'Watch a Movie'],
+};
 
 const Habitoptions = ({ navigation }) => {
-  const [options, setOptions] = useState([
-    { id: 1, name: "Exercise" },
-    { id: 2, name: "Morning Stretching" },
-    { id: 3, name: "Walk" },
-    { id: 4, name: "Eat Healthy" },
-    { id: 5, name: "Wake Up Early" },
-    { id: 6, name: "Read" },
-    { id: 7, name: "Write in Journal" },
-    { id: 8, name: "Meditate" },
-    { id: 9, name: "Limited Screen Time" },
-    { id: 10, name: "Compleate Daily To-Do List" },
-    { id: 11, name: "Study" },
-    { id: 12, name: "Spend Time with Family" },
-    { id: 13, name: "Connect with Friends:" },
-    { id: 14, name: "Express Gratitude" },
-    { id: 15, name: "Networking" },
-    { id: 16, name: "Cooking" },
-    { id: 17, name: "Outdoor Time" },
-    { id: 18, name: "Watch a Movie" },
-    
-  ]);
   const [newHabit, setNewHabit] = useState('');
-
-  const handleSelectOption = (option) => {
-    navigation.navigate('Home', { selectedHabit: option });
-  };
+  const [expandedCategory, setExpandedCategory] = useState('');
 
   const addNewHabit = () => {
     if (newHabit.trim() === '') return;
-
-    const newId = options.length + 1;
-    setOptions([...options, { id: newId, name: newHabit }]);
+    // Add new habit to a default or specific category
+    habitCategories['Personal Growth'].push(newHabit.trim());
     setNewHabit('');
+    // Optionally, navigate to home screen with the new habit
```

Comment:
you shouldn't leave out these type of comments, type take up unnecessary space.

Reply:
Done